cli/search/search.go

```diff
@@ -115,6 +116,54 @@ func GetVersionsFromGitRemote(repo string) ([]version.Version, error) {
 	return versions, nil
 }
 
+// GetCommitsFromGitLocal returns commits list from specified local git repo.
+func GetCommitsFromGitLocal(repo string, hashLenght int) ([]string, error) {
+	if _, err := exec.LookPath("git"); err != nil {
+		return nil, fmt.Errorf("unable to get commits: `git` command is missing")
+	}
+
+	abbrevStr := "--abbrev=" + strconv.Itoa(hashLenght)
+
+	cmd := exec.Command("git", "log", "--format=%h", abbrevStr)
+	cmd.Dir = repo
+
+	output, _ := cmd.Output()
+
+	lines := strings.Split(strings.TrimSpace(string(output)), "\n")
+
+	// No tags found.
+	if len(lines) == 1 && lines[0] == "" {
+		return []string{}, nil
+	}
+
+	return lines, nil
+}
+
+// GetCommitsFromGitRemote returns commits list from specified remote git repo.
+func GetCommitsFromGitRemote(repo string, hashLenght int) ([]string, error) {
+
+	if _, err := exec.LookPath("git"); err != nil {
+		return nil, fmt.Errorf("unable to get commits: `git` command is missing")
+	}
+
+	tempRepoPath, err := os.MkdirTemp("", "versions_log")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commits from %q: %s", repo, err)
+	}
+
+	defer os.RemoveAll(tempRepoPath)
+
+	cmd := exec.Command("git", "clone", "--filter=blob:none", "--no-checkout",
+		"--single-branch", repo, tempRepoPath)
```

Comment:
Can we support something like the following?

```sh
$ tt install tt pr/534
```

I can do it using git manually.

```sh
$ cd path/to/tt
$ git fetch origin pull/534/head:pr/534
<...>
From github.com:tarantool/tt
 * [new ref]         refs/pull/534/head -> pr/534
$ git checkout pr/534
Switched to branch 'pr/534'
$ git log | head -n 1
commit efd4f08f8ff8a092479120b3218bfbdea379018b
```

See also https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally#modifying-an-inactive-pull-request-locally and https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Github_pull_request_made_easy.

Reply:
Thank you for the suggestion. Support for installation via pr id has been added with `tt install tt/tarantool pr/<ID>` command

Reply:
Thanks!